Solinea/goldstone-server
# Flatten compliance to submodule

This PR replaces the previous addon management strategy with one based on git submodules, and an assumption that we know and control the universe of addons.  That assumption should be fairly easy to relax later, but seemed acceptable for the foreseeable future.

In addition, it incorporates #146 and extends it to provide REST API for SavedSearch operations.  There will be a parallel PR in the goldstone-compliance project that should be reviewed and tested at the same time.

Finally, there were some updates to logstash version (now 2.1) and rule handling to better support TRACEs, and API performance tracking.

To test:
* pull this branch and check it out
* `git submodule init`
* `git submodule update`
* `cd goldstone/compliance`
* `git checkout flatten_to_submodule`
* `cd ../..`
* `docker rm -f $(docker ps -qa)`
* `docker rmi goldstoneserver_gsappdev goldstoneserver_gsdbdvc goldstoneserver_gssearch goldstoneserver_gslog`
* run test suite
* start client and poke around (should see compliance addon installed now)
* test the /core/saved_search API and the /compliance/defined_search API
** should be 12 saved_searches
** should be 10 defined_searches

closes: #146
closes: #188
closes: #127
